src/Aquifer.Jobs/Subscribers/ResourceContentVersionSimilarityMessageSubscriber.cs

```diff
@@ -0,0 +1,239 @@
+using Aquifer.Common.Messages;
+using Aquifer.Common.Messages.Models;
+using Aquifer.Common.Tiptap;
+using Aquifer.Common.Utilities;
+using Aquifer.Data;
+using Aquifer.Data.Entities;
+using Azure.Storage.Queues.Models;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Aquifer.Jobs.Subscribers;
+
+public sealed class ResourceContentVersionSimilarityMessageSubscriber(
+    AquiferDbContext _dbContext,
+    ILogger<ResourceContentVersionSimilarityMessageSubscriber> _logger)
+{
+    [Function(nameof(ScoreResourceContentVersionSimilarityMessageSubscriber))]
+    public async Task ScoreResourceContentVersionSimilarityMessageSubscriber(
+        [QueueTrigger(Queues.GenerateResourceContentSimilarityScore)]
+        QueueMessage queueMessage,
+        CancellationToken ct)
+    {
+        await queueMessage.ProcessAsync<ScoreResourceContentVersionSimilarityMessage, ResourceContentVersionSimilarityMessageSubscriber>(
+            _logger,
+            nameof(ScoreResourceContentVersionSimilarityMessageSubscriber),
+            ProcessAsync,
+            ct);
+    }
+
+    private async Task ProcessAsync(QueueMessage queueMessage, ScoreResourceContentVersionSimilarityMessage message, CancellationToken ct)
+    {
+        var similarityScoreEntity = await GenerateResourceContentVersionSimilarityScoreEntity(message, _dbContext, ct);
+        
+        await _dbContext.ResourceContentVersionSimilarityScores.AddAsync(similarityScoreEntity, ct);
+        await _dbContext.SaveChangesAsync(ct);
+    }
+    
+    private async Task<ResourceContentVersionSimilarityScoreEntity> GenerateResourceContentVersionSimilarityScoreEntity(
+        ScoreResourceContentVersionSimilarityMessage message,
+        AquiferDbContext dbContext,
```

Comment:
I suggest using the class member's `_dbContent` instead of passing it here.  The same suggestion applies to the `private static` methods below doing the same thing but they'll need to be made non-static as well.